### PR TITLE
Feat: Add new config "self" to submit to CE from within a cluster

### PR DESCRIPTION
**Pull request by SebastianMorawiec**
https://zapatacomputing.atlassian.net/browse/ORQSDK-772

# The problem

Clusters have embedded passport token to be used from within their internals.
For user convenience they shouldn't have to log-in into the same cluster from within portal studio

# This PR's solution

add new config named <self> which can be used to interact with CE from within portal and studio.

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/internal/docs>[!]:`

### CODE REVIEW

**Review by alexjuda — CHANGES_REQUESTED** [review]

File: src/orquestra/sdk/_base/_config.py [hunk 1]
```diff
@@ -35,6 +36,7 @@
 BUILT_IN_CONFIG_NAME = "local"
 RAY_CONFIG_NAME_ALIAS = "ray"
 IN_PROCESS_CONFIG_NAME = "in_process"
+SAME_CLUSTER_CONFIG_NAME = "self"
```
[the remark this answers is not in the document]
Comment (SebastianMorawiec) [reply src/orquestra/sdk/_base/_config.py:39]: There was a thread on slack: https://zapatista.slack.com/archives/C02V25SBGS3/p1681472324307459?thread_ts=1681466373.464069&cid=C02V25SBGS3 where we discussed it.
We can get back to it anyway if you want offline

**Comment by alexjuda** [comment]
Related to: https://zapatacomputing.atlassian.net/browse/ORQSDK-772

File: src/orquestra/sdk/_base/_config.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexjuda) [reply src/orquestra/sdk/_base/_config.py:39]: Extracted into:
* https://zapatacomputing.atlassian.net/browse/ORQSDK-818
* https://zapatacomputing.atlassian.net/browse/ORQSDK-819

**Review by alexjuda — APPROVED** [review]
LGTM, with some non-blocking suggestions :)

**Pull request merged**